python/audit/middleware.py

```diff
@@ -10,30 +10,22 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import json
+from django.contrib.auth.models import AnonymousUser, User
 
 from audit.models import AuditSession, AuditSessionLoginResult
-from rbac.models import User
 
 
 class AuditLoginMiddleware:
     def __init__(self, get_response):
         self.get_response = get_response
 
-    def __call__(self, request):
-        if not request.path == '/api/v1/rbac/token/':
-            return self.get_response(request)
-
-        body = json.loads(request.body.decode('utf-8'))
-        username = body['username']
-        response = self.get_response(request)
-
-        if request.user.is_authenticated:
-            audit_user = request.user
+    @staticmethod
+    def _audit(user: User | AnonymousUser = None, username: str = None):
+        """Authentication audit"""
+        if user.is_authenticated:
```

Comment:
If `user` is `None` `AttributeError` will be raised

Reply:
fixed